styleguide/core/Input.hbs

```diff
@@ -22,8 +22,8 @@
         <div class="{{blockName}}"{{extraAttributes}}>
 
             {{#defineBlockBody}}
-                {{element "input"}}
                 {{element "label"}}
+                {{element "input"}}
```

Comment:
I do think a template per input type would fix this, and other issues. While it would be duplicating some code, in the end, I think it would make things simpler and easier to understand